src/modules/samples/actions.js

```diff
@@ -16,17 +19,39 @@ export const get = id => async (dispatch, getState) => {
     await dispatch(networkAction(GET, api.samples.get(id), { meta: { id } }));
 };
 
+export const setFilter = (name, value) => {
+    return {
+        type: SET_FILTER,
+        data: { name, value }
+    }
+};
+
+export const clearFilters = () => {
+    return {
+        type: CLEAR_FILTERS,
+    }
+};
+
 export const list = ({ offset = 0, limit = DEFAULT_PAGINATION_LIMIT } = {}) => async (dispatch, getState) => {
     if (getState().samples.isFetching) return;
 
-    const pageOptions = { limit, offset }
+    const filters = getState().samples.filters
+    const options = { limit, offset, ...filters}
 
     await dispatch(networkAction(LIST,
-        api.samples.list(pageOptions),
-        { meta: pageOptions }
+        api.samples.list(options),
+        { meta: options }
     ));
 };
 
+export const listBiospecimenTypes = () => async (dispatch, getState) => {
+    // Check if we're already fetching or have fetched sample biospecimen types first (they won't change dynamically.)
+    if (getState().sampleBiospecimenTypes.isFetching || getState().sampleBiospecimenTypes.items.length > 0)
+        return;
```

Comment:
Superfluous comment imho, the `isFetching` check is present in most if not all network actions.
Also I would remove the `length > 0` check.